src/Symfony/Component/Form/ChoiceList/Factory/PropertyAccessDecorator.php

```diff
@@ -81,19 +85,36 @@ public function createListFromChoices(iterable $choices, $value = null)
             };
         }
 
-        return $this->decoratedFactory->createListFromChoices($choices, $value);
+        if (\is_string($filter)) {
+            $filter = new PropertyPath($filter);
+        }
+
+        if ($filter instanceof PropertyPath) {
+            $accessor = $this->propertyAccessor;
+            $filter = static function ($choice) use ($accessor, $filter) {
+                if (\is_object($choice) || \is_array($choice)) {
+                    return (bool) $accessor->getValue($choice, $filter);
```

Comment:
shouldn't the "if" be replaced by this?:
`return (\is_object($choice) || \is_array($choice) && $accessor->getValue($choice, $filter);`
(same in other similar places)

Reply:
Right, done